Puara/Shake.cpp

```diff
@@ -3,7 +3,50 @@ namespace puara_gestures::objects
 {
 void Shake::operator()(halp::tick t)
 {
-  // TODO
-  // outputs.output = impl.shake(inputs.accel, inputs.gyro, inputs.mag, 0.1);
+
+  // getting input data
+
+  const puara_gestures::Coord3D& current_accel = inputs.accel.value;
+  const float desired_frequency_hz = inputs.integrator_frequency.value;
+  const float desired_fast_leak = inputs.fast_leak_param.value;
+  const float desired_slow_leak = inputs.slow_leak_param.value;
+  const float desired_activation_threshold = inputs.activation_threshold_param.value;
+
+  // updating puara parameters
+
+  //checking if the desired frequency for the integrator has changed
+  int desired_frequency_int = static_cast<int>(desired_frequency_hz);
```

Comment:
please do not static_cast every time there is a value conversion, C++ has implicit value conversions for a reason ^^'

Reply:
and if you have a linter / IDE that throws you a warning for this, this is a braindead warning that you ought to disable